File: src/main/reactjs/src/components/Product.js
import ProductOption from "./ProductOption.js";

function Product({product}) {
    return (
        <div className="product-figure">
            <div className="item-info">
                <span>{product.categoryName}</span>
                <span className="item-name">{product.productName}</span>
                <div className="item-options">
                    {
                        product.productOptions.map(option => {
                            return <ProductOption key={option.id} option={option}></ProductOption>
                        })
                    }
                </div>
            </div>
            <div className="item-details">
                <span>{product.productName}</span>
                <span>{product.productDesc}</span>
                <span className="close"></span>
            </div>
        </div>
    )
}

export default Product;

File: src/main/java/com/doran/doran/model/entity/Product.java
package com.doran.doran.model.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "PRODUCT")
public class Product {
    @Id
    @GeneratedValue
    private int productId;
    private String productName;
    private String productDesc;
    private String productImg;
    private Boolean productAvailable;
    @OneToMany(mappedBy = "product")
    @JsonManagedReference
    private List<ProductOption> options = new ArrayList<>();
    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "categoryId")
    private Category category;

    public void addOptions(ProductOption option) {
        options.add(option);
        option.setProduct(this);
    }
}
